pallets/file-system/src/lib.rs

```diff
@@ -313,12 +394,27 @@ pub mod pallet {
             let start_block = NextStartingBlockToCleanUp::<T>::get();
             let mut block_to_clean = start_block;
 
-            // Total weight used to avoid exceeding the remaining weight
+            // Total weight used to avoid exceeding the remaining weight.
+            // We count one write for the `NextBlockToCleanup` storage item updated at the end.
             let mut total_used_weight = Weight::zero();
 
             let required_weight_for_iteration =
                 db_weight.reads_writes(1, T::MaxExpiredStorageRequests::get().into());
 
+            // Check if we can process a single iteration and update the `NextStartingBlockToCleanUp` storage item
+            if !remaining_weight.all_gte(
+                match total_used_weight.checked_add(&required_weight_for_iteration) {
+                    Some(weight) => {
+                        // Update `total_used_weight` to take into account the final write operation
+                        total_used_weight = db_weight.writes(1);
```

Comment:
Moreover, this increment feels weird that it is here. It would be more readable if it were closer to the reason why it is added in the first place. Here:
```rust
            // `NextStartingBlockToCleanUp` is always updated to start from the block we reached in the current `on_idle` call.
            if block_to_clean > start_block {
                NextStartingBlockToCleanUp::<T>::put(block_to_clean);
                // Add it here
            }
```

Reply:
As mentioned above ^, if we add a write here without checking at the beginning of the `on_idle` execution, we cannot be sure that the write will not have the `total_used_weight` exceed the `remaining_weight`.